### PR TITLE
[PKG-1208] catboost 1.2

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Changelog: https://github.com/catboost/catboost/releases
License: https://github.com/catboost/catboost/blob/v1.2/LICENSE
Requirements:
- https://github.com/catboost/catboost/tree/v1.2/catboost/python-package
- https://github.com/catboost/catboost/blob/v1.2/CMakeLists.txt
- https://github.com/catboost/catboost/blob/v1.2/catboost/python-package/pyproject.toml
- https://github.com/catboost/catboost/blob/v1.2/catboost/python-package/setup.py
- https://github.com/catboost/catboost/blob/v1.2/catboost/python-package/CMakeLists.txt

Actions:
1. Update pypi wheel links in` build.sh`
2. Add `--no-build-isolation`
3. Add a workaround for renaming wheel files for `osx-64`
4. Skip **python 3.8** on `osx-arm64`
5. Skip `ppc64le` and `s390x`
6. Add `curl` for downloading wheel files
7. Fix `home` url
8. Fix `license_family`
9. Update `doc_url`
10. Add a comment about `python-graphviz` usage
11. Move `test/imports` and `commands` to `run_test.py` and add `pip check` + catboost examples to verify that it works correctly, especially on osx-64

### CODE REVIEW

**Review by akabanovs — APPROVED** [review]
LGTM
Where did you find the test you added to the recipe?

**Comment by skupr-anaconda** [comment]
> LGTM Where did you find the test you added to the recipe?

From the docs https://catboost.ai/en/docs/concepts/python-quickstart

File: recipe/run_test.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import subprocess
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/run_test.py:1]: this is new file, where did you get it from and what it actually test ? I do not see any pytest asserts here, just wondering ?

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

File: recipe/run_test.py (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/run_test.py:1]: I created it myself just to check if catboost is working and added there pip check too

**Review by varlackc — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
-  skip: True  # [linux and not x86_64]
+  # - Linux: Python 3.8, 3.9, 3.10, 3.11
+  # - OS X: Python 3.8, 3.9, 3.10, 3.11
+  # - Win-64: Python 3.8, 3.9, 3.10, 3.11
+  # There are issues with python 3.8 on macOS 11.x: https://github.com/conda-forge/python-feedstock/issues/445
+  #  so Conda has issues with tags compatibility with platform tags 'macosx_11_0_...'
+  #  so disable this configuration for now
+  skip: True  # [(osx and x86_64) and py==38]
+  # TODO: support Linux on ppc64le: https://github.com/catboost/catboost/issues/2145 and s390x
+  skip: True  # [ppc64le or s390x]
```
Comment (boldorider4) [inline recipe/meta.yaml:19]: why are we skipping s390x?

Comment (skupr-anaconda) [reply recipe/meta.yaml:19]: Missing wheels for this platform:
`ERROR: catboost-1.2-cp38-cp38-manylinux2014_.whl is not a supported wheel on this platform.`

Comment (boldorider4) [reply recipe/meta.yaml:19]: is it the same for all py versions?

Comment (skupr-anaconda) [reply recipe/meta.yaml:19]: No, for example, catboost-1.2-cp311-cp311-**manylinux2014_aarch64**.whl is for `linux-aarch64`.
There are missing wheels for s390x https://pypi.org/project/catboost/1.2/#files

Comment (skupr-anaconda) [reply recipe/meta.yaml:19]: Ah, there is no value for s390x https://github.com/AnacondaRecipes/catboost-feedstock/pull/4/files#diff-d7075654874cb08007a21aaab3ecd4b3453a9087e7505d034d548b8938b599bcR41-R44

**Pull request merged**